controllers/admin/AdminAjaxPs_buybuttonliteController.php

```diff
@@ -0,0 +1,118 @@
+        );
+        $sql->leftJoin('product_attribute', 'pa', 'pa.`id_product` = p.`id_product`');
+        $where = 'pl.`name` LIKE \'%' . pSQL($query) . '%\'
+        OR p.`reference` LIKE \'%' . pSQL($query) . '%\'
+        OR EXISTS(SELECT * FROM `' . _DB_PREFIX_ . 'product_supplier` sp WHERE sp.`id_product` = p.`id_product` AND `product_supplier_reference` LIKE \'%' . pSQL($query) . '%\')';
+        $sql->orderBy('pl.`name` ASC');
+        if (Combination::isFeatureActive()) {
+            $where .= ' OR EXISTS(SELECT * FROM `' . _DB_PREFIX_ . 'product_attribute` `pa` WHERE pa.`id_product` = p.`id_product` AND (pa.`reference` LIKE \'%' . pSQL($query) . '%\'))';
+        }
+        $sql->where($where);
+        $result = Db::getInstance()->executeS($sql);
+        if (!$result) {
+            return false;
+        }
+        $results_array = array();
+        foreach ($result as $row) {
+            if ($this->hasCombinations($row['id_product'])) {
+                $row['attribute_name'] = $this->getAttributes($row['id_product_attribute'], $id_lang);
+                $row['image_link'] = $this->getProductAttributeImage($row['id_product'], $row['id_product_attribute']);
+            } else {
+                $row['image_link'] = $this->getProductImage($row['id_product']);
+            }
+            $results_array[] = $row;
+        }
+        $this->content = json_encode($results_array);
+    }
+
+    public function getProductImage($id_product)
+    {
+        $product = new Product($id_product);
+        $link_rewrite = $product->link_rewrite;
+        $link = new Link();
+        $id_image = Image::getCover($id_product);
+        $id_image = $id_image['id_image'];
+
+        $image_link = $link->getImageLink($link_rewrite, $id_image, ImageType::getFormattedName('small'));
+
+        return Tools::getProtocol().$image_link;
+    }
+
+    public function getProductAttributeImage($id_product, $id_product_attribute)
+    {
+        $context = Context::getContext();
+        $id_lang = $context->language->id;
+        $id_shop = $context->shop->id;
+
+        $product = new Product($id_product);
+        $link_rewrite = $product->link_rewrite;
+        $link = new Link();
+        $id_image = Image::getBestImageAttribute($id_shop, $id_lang, $id_product, $id_product_attribute);
+        $id_image = $id_image['id_image'];
+
+        $image_link = $link->getImageLink($link_rewrite, $id_image, ImageType::getFormattedName('small'));
+
+        return Tools::getProtocol().$image_link;
+    }
+
+    public function hasCombinations($id_product)
+    {
+        if (is_null($id_product) || 0 >= $id_product) {
+            return false;
+        }
+        $attributes = Product::getAttributesInformationsByProduct($id_product);
+        return !empty($attributes);
+    }
+
+    public function getAttributes($id_product_attribute, $id_lang)
+    {
+        $sql = 'SELECT agl.name as label, al.name as value
```

Comment:
👍 This SQL query is a lot easier to read thanks to the nice way it is written

Bonus: table aliases `al`, `ag` and `agl` are hard to distinguish, maybe use more meaningful names ?